pi_finder/package/finder.js

```diff
@@ -42,14 +43,44 @@ proto.whitelist = [
 ];
 
 proto.start = function(cb) {
+  //get all interfaces
+  var interfaces = os.networkInterfaces();
+  var subnets = [];
+  //pull out subnets for IPv4 interfaces only
+  for(var name in interfaces) {
+    var ifc = interfaces[name];
+    for(var i=0; i<ifc.length; i++) {
+      if(ifc[i].family == 'IPv4') {
+        subnets.push(ifc[i].address);
+      }
+    }
+  }
+  //filter out loopback
+  subnets = subnets.filter(function(n) { return n !== '127.0.0.1'; });
+  this.total_ips = [];
+  this.doSubnet(subnets,cb);
+};
 
+//recursive function to process each subnet
+proto.doSubnet = function(subnets,cb) {
+  var addr = subnets.shift();
+  this.subnet = addr.substr(0, addr.lastIndexOf('.')) || false;
```

Comment:
I might tweak this a bit in the future since we no longer are just using one subnet.